### ISSUE TITLE
UserLand has entered an illegal state couldn't extract filesystem

### ISSUE THREAD

**Issue by Appeee** [CypherpunkArmory/UserLAnd#703]
After downloading the assets UserLand quits with the message in the title. Clearing the support files and reinstalling the app does not help. Log attached. "proot info: vpid 1: terminated with signal 11"

# Device Information

Device: Sony Xperia Z3
Android Version: Oreo 8.1
Kernel: 3.4.0-lineage-g784c3f5b 
UserLAnd Version 2.3.8  
The device is rooted with Magisk 18.1
Running CarbonRom CR-6.1 - 20190311-1839
[PRoot_Debug_Log.log](https://github.com/CypherpunkArmory/UserLAnd/files/2970292/PRoot_Debug_Log.log)

# Additional context

**Comment by MatthewTighe** [comment]
We recently discovered our download process can cause files to become corrupted, as we are currently downloading our assets directly from a git tree. We're working on an upgrade to the process that will hopefully ship before the end of next week that will fix issues like these. My first assumption is that this is what is causing your issue.

Any other thoughts @corbinlc ?

**Comment by lucipher666** [comment]
Im getting the same message on my alcatel a574bl , Android version 7.1.1, Kernal 3.10.49-g9d0253d unrooted

**Comment by rbthomp** [comment]
I'd just like to add that I'm getting the same error, but mine originated after restarting the phone several times. After words I couldn't get my session to start. I then proceeded to remove the filesystem and recreate it when I received the error. I have had this issue on 4 Galaxy S6 Android version 7.0. for whatever reason debug logging dose not create the logging file.

**Comment by Appeee** [comment]
Hi there,

some progress might have arose cause now it completes the downloads, copies the assetts and only after verifying it pops up the same error message. Hope it helps. A.

**Comment by Areahints** [comment]
Similar issue. But mine was working now suddenly it never passes <illegal state>

**Comment by MatthewTighe** [comment]
Do you have enough internal storage space to extract a filesystem? You should plan to have at least 1GB free. We realized recently that not the process will still be attempted even without the requisite space and fail without a useful error message. We're working on correcting that.

**Comment by Areahints** [comment]
I'm just curious but do all the Linux distributions require 1GB minimum?

**Comment by MatthewTighe** [comment]
No, but they will quickly reach that size as soon as they are used for anything. Even `apt-get update && apt-get upgrade` tend to balloon size substantially.

**Comment by pferdefreund** [comment]
after apt-get update && apt-get upgrade you should use apt-get clean to remove the installed packages from /var/apt/cache. Then you have no balloon-effect.

**Comment by davidadriao** [comment]
Changing the userland app from sd card to internal memory fixed this for me.

**Comment by Marchantia1** [comment]
I have a similar issue as the OP on a rooted Fairphone running LineageOS 15.1, UserLAnd 2.5.13, for any distro I try to install. The error message also says "Failure reason: [Command failed with: 255]" and after dismissing that dialog the app stays stuck on "Setting up filesystem... Extracting: ⛝[H⛝[J". (It's not exactly ⛝ but looks like it.)

Clearing support files does not help; nor does fully deleting UserLAnd (including its data) and reinstalling it.

I tried to create a PRoot_Debug_Log file but I can't seem to find it in /mnt/sdcard/

Things used to work some months back; don't remember with which version of UserLAnd that was.

**Comment by Keskebeu** [comment]
Just hit the three dots on top corner and clear system file. It will work after. This happens when the download has been disrupted or canceled while in process. 

Hope this helps, it helped me.

**Comment by pferdefreund** [comment]
at the moment everything works fine. Since the last update no problems anymore. Thank you and greetings from Germany

Klaus

Am 3. Juli 2019 04:34:52 MESZ schrieb Keskebeu <notifications@github.com>:

**Comment by corbinlc** [comment]
I think these issue have been solved.  If you are still seeing this, please create a new issue with information specific to what device you are using and what behavior you are observing.

**Comment by xxzahhxx** [comment]
**it does this everytime i try to launch a custom session**

**Comment by xxzahhxx** [comment]
**if anyone sees this in 2024 that is**
xxzahhxx@gmail.com

**Comment by Rusna** [comment]
How to resolve?
